yarn-project/end-to-end/src/e2e_slow_tree.test.ts

```diff
@@ -0,0 +1,146 @@
+
+    constructor(circuitsWasm: CircuitsWasm) {
+      this.circuitsWasm = circuitsWasm;
+    }
+    compressInputs(_inputs: Buffer[]): Buffer {
+      throw new Error('Method not implemented.');
+    }
+    hashToField(_data: Uint8Array): Buffer {
+      throw new Error('Method not implemented.');
+    }
+    hashToTree(_leaves: Buffer[]): Promise<Buffer[]> {
+      throw new Error('Method not implemented.');
+    }
+
+    public compress(lhs: Buffer, rhs: Buffer): Buffer {
+      return pedersenPlookupCommitInputs(this.circuitsWasm, [lhs, rhs]);
+    }
+  }
+
+  it('Messing around with noir slow tree', async () => {
+    const circuitsWasm = await CircuitsWasm.get();
+    const hasher = new PedersenHasher(circuitsWasm);
+
+    const db = levelup(createMemDown());
+    const depth = 254;
+    const tree = await newTree(SparseTree, db, hasher, 'test', depth);
+    const root = tree.getRoot(true);
+
+    const getMembershipProof = async (index: bigint, includeUncommitted: boolean) => {
+      return {
+        index,
+        value: Fr.fromBuffer((await tree.getLeafValue(index, includeUncommitted))!),
+        // eslint-disable-next-line camelcase
+        sibling_path: (await tree.getSiblingPath(index, includeUncommitted)).toFieldArray(),
+      };
+    };
+
+    const getUpdateProof = async (newValue: bigint, index: bigint) => {
+      const beforeProof = await getMembershipProof(index, false);
+      const afterProof = await getMembershipProof(index, true);
+
+      return {
+        index,
+        // eslint-disable-next-line camelcase
+        new_value: newValue,
+        // eslint-disable-next-line camelcase
+        before: { value: beforeProof.value, sibling_path: beforeProof.sibling_path },
+        // eslint-disable-next-line camelcase
+        after: { value: afterProof.value, sibling_path: afterProof.sibling_path },
+      };
+    };
+
+    const status = async (key: bigint) => {
+      logger(`\tTime: ${await cheatCodes.eth.timestamp()}`);
+      logger('\tRoot', await contract.methods.un_read_root(owner).view());
+      logger(`\tLeaf (${key})`, await contract.methods.un_read_leaf_at(owner, key).view());
+    };
+
+    const owner = wallet.getCompleteAddress().address;
+    const key = owner.toBigInt();
+
+    await contract.methods.initialize(root).send().wait();
+
+    logger('Initial state');
+    await status(key);
+
+    await contract.methods
+      .read_at({ ...(await getMembershipProof(key, true)), value: 0n })
+      .send()
+      .wait();
+
+    logger(`Updating tree[${key}] to 1 from public`);
+    await contract.methods
+      .update_at(await getUpdateProof(1n, key))
+      .send()
+      .wait();
+    await tree.updateLeaf(new Fr(1).toBuffer(), key);
+    await status(key);
+
+    const zeroProof = await getMembershipProof(key, false);
+    logger(`"Reads" tree[${zeroProof.index}] from the tree, equal to ${zeroProof.value}`);
+    await contract.methods.read_at(zeroProof).send().wait();
```

Comment:
why do you not need to pass in `value` here but in L103 you do?

Reply:
In L103 I was just overwriting the value to make it explicit.